### PR TITLE
gh-84559: Change the multiprocessing start method default to `forkserver`

### PR DESCRIPTION
**Pull request by gpshead**
We change the default from `fork` to `forkserver` or `spawn`. See the issue for details.

This change is intended for Python 3.14.

TODO:

 * [x] The 3.13 release branch has been made, thus starting 3.14 in `main`.
 * [x] NEWS entry added.
 * Documentation updates in:
    * [x] library/multiprocessing.rst
    * [x] library/concurrent.futures.rst
    * [x] whatsnew/3.14.rst

* Issue: gh-84559

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2024-09-19-00-09-48.gh-issue-84559.IrxvQe.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+The default :mod:`multiprocessing` start method on Linux and other POSIX systems has been changed away from often unsafe``"fork"`` to ``"forkserver"`` (when the platform support sending file handles over pipes) or ``"spawn"``. Mac and Windows are unchanged as they already default to ``"spawn"``.
```
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2024-09-19-00-09-48.gh-issue-84559.IrxvQe.rst]: Lint fix:
```suggestion
The default :mod:`multiprocessing` start method on Linux and other POSIX systems has been changed away from often unsafe ``"fork"`` to ``"forkserver"`` (when the platform support sending file handles over pipes) or ``"spawn"``. Mac and Windows are unchanged as they already default to ``"spawn"``.
```

**Comment by gpshead** [comment]
@ericsnowcurrently could you review? (likely an easy one)

File: Lib/test/support/__init__.py [hunk 2]
```diff
@@ -2209,7 +2209,15 @@ def skip_if_broken_multiprocessing_synchronize():
...
             # a file in /dev/shm/ directory.
-            synchronize.Lock(ctx=None)
+            import multiprocessing
+            synchronize.Lock(ctx=multiprocessing.get_context('fork'))
+            # The explicit fork mp context is required in order for
+            # TestResourceTracker.test_resource_tracker_reused to work.
+            # synchronize creates a new multiprocessing.resource_tracker
+            # process at module import time via the above call in that
+            # scenario. Awkward. This enables gh-84559. No code involved
+            # should have threads at that point so fork() should be safe.
```
Comment (ericsnowcurrently) [inline Lib/test/support/__init__.py:2219]: "Awkward."  LOL

(Good comment overall too.)

File: Lib/test/_test_multiprocessing.py [hunk 3]
```diff
@@ -5553,15 +5553,29 @@ def test_set_get(self):
...
+            self.assertIn('fork', methods)
+            if other_methods := set(methods) - {'fork', 'spawn'}:
+                # If there are more than those two, forkserver must be one.
+                self.assertEqual({'forkserver'}, other_methods)
+            # The default is the first method in the list.
+            self.assertIn(methods[0], {'forkserver', 'spawn'},
+                          msg='3.14+ default must not be fork')
+            if methods[0] == 'spawn':
+                # Confirm that the current default selection logic prefers
+                # forkserver vs spawn when available.
```
Comment (ericsnowcurrently) [inline Lib/test/_test_multiprocessing.py:5577]: That's a good check!

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

Thanks for following this through!

File: Lib/test/support/__init__.py (hunk 2 above)
Comment (gpshead) [reply Lib/test/support/__init__.py:2219]: thanks!  yeah... my previous wording was even more confusing. i was glad i reread what i wrote ages ago last night and reworded it to what you see here. :)

**Pull request merged**